src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -104,10 +104,12 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
             marshaller.marshal(toSave, fileWriter);
 
-        } catch (IOException ioe) {
-            throw new StorageOperationException("Error writing to file: " + path);
         } catch (JAXBException jaxbe) {
             throw new StorageOperationException("Error converting address book into storage format");
+        } catch (FileNotFoundException fnfe) {
```

Comment:
This is incorrect. The original code is written such that it always re-save the file, i.e. the missing file situation will never be detected.

You are supposed to check for the existence of the file and raise the exception if file is missing _before_ the actual writing :-).